extensions/github1s/src/api.ts

```diff
@@ -82,25 +63,33 @@ export const validateToken = (token: string) => {
 	});
 };
 
-const branchNameCache = new Map();
-
-export const fetchGitHubBranches = (owner: string, repo: string) => {
-	const key = owner + '/' + repo;
-	if (branchNameCache.has(key)) {
-		return Promise.resolve(branchNameCache.get(key));
-	}
-	return fetch(`https://api.github.com/repos/${owner}/${repo}/branches`)
-		.then(res => {
-			const names = res.map(x => x.name);
-			branchNameCache.set(key, names);
-			return names;
+export const getGithubBranches = (owner: string, repo: string) => {
+	return fetch(`https://api.github.com/repos/${owner}/${repo}/branches?per_page=100`)
+		.then(branches => {
+			// TODO: only no more than 200 branches are supported
+			if (branches.length === 100) {
```

Comment:
Why not just support maximum 100 branches? I believe it's normal for majority repos.